src/librustc_metadata/cstore_impl.rs

```diff
@@ -210,6 +210,23 @@ provide! { <'tcx> tcx, def_id, other, cdata,
     }
     defined_lang_items => { Rc::new(cdata.get_lang_items(&tcx.dep_graph)) }
     missing_lang_items => { Rc::new(cdata.get_missing_lang_items(&tcx.dep_graph)) }
+
+    item_body => {
+        if let Some(cached) = tcx.hir.get_inlined_body_untracked(def_id) {
+            return cached;
+        }
```

Comment:
You can remove this `if` and the `get_inlined_body_untracked` method, as the result will be cached anyway.